sympy/solvers/solveset.py

```diff
@@ -271,6 +273,63 @@ def _invert_complex(f, g_ys, symbol):
     return (f, g_ys)
 
 
+def invert_modular(f_x, y, m, x, domain=S.Integers):
+    """Inverts a modular equation"""
+    x = sympify(x)
+    if not x.is_Symbol:
+        raise ValueError("x must be a symbol")
+    f_x = sympify(f_x)
+    if not f_x.has(x):
+        raise ValueError("Inverse of constant function doesn't exist")
+    y = sympify(y)
+    if y.has(x):
+        raise ValueError("y should be independent of x ")
+
+    if domain.is_subset(S.Integers):
+        x1, s = _invert_modular(f_x, FiniteSet(y), m, x)
```

Comment:
What if domain is not a subset of Integers? As written, this will attempt to return x1 which has not been defined. Maybe raise an exception?